src/util/addDecorationsToEditor.ts

```diff
@@ -169,3 +168,33 @@ export function addDecorationsToEditors(
     });
   });
 }
+
+interface Lexeme {
+  /** The normalised text of the lexeme. */
+  text: string;
+
+  /** The start offset of the lexeme within its containing token */
+  tokenStartOffset: number;
+
+  /** The end offset of the lexeme within its containing token */
+  tokenEndOffset: number;
+}
+
+function getTokenLexemes(
```

Comment:
Note how this generic function will make it easier to handle unicode with diacritics, as well as #81